provider/read.go

```diff
@@ -65,7 +65,7 @@ func (s *RawProviderServer) ReadResource(ctx context.Context, req *tfprotov5.Rea
 		return resp, nil
 	}
 	co := resState["object"]
-	cu, err := payload.FromTFValue(co, tftypes.AttributePath{})
+	cu, err := payload.FromTFValue(co, tftypes.NewAttributePath())
```

Comment:
This isn't related to this PR, but just something I noticed on line 59. I think you might want a `if err != nil {` there too, to prevent a null pointer dereference with `err.Error()`. Here's a quick test to demonstrate that one:

```
gore> var err error
gore> err = nil
nil
gore> err.Error()
panic: runtime error: invalid memory address or nil pointer dereference
```

Granted, there's a lot I don't understand about how `err` will get its value, but I did notice that there are several cases inside of `currentState.As()` that return nil. (Sorry this is all like linter stuff... it's because I'm unfamiliar with the codebase, so linting+testing is the best I can do for review).

Reply:
That's another good catch! You're right, `err.Error()` on line 63 can panic. This smells like a copy-paste error where I got the contents on the block starting at line 59 from the block right above it  that was actually checks err for nil.

Which linter are you using BTW? We should incorporate in the `make test` workflow.

Reply:
I re-arranged the checks there and removed the use or `err.Error()` since it was handled right before this and if the execution made it all the way here, it's guaranteed that `err` is nil at this point (because it would have returned early a few lines above if it weren't).

Reply:
I was just using my IDE, Goland, but `golangci-lint` seems to catch some things too. Eventually we can get it into CI. There's an [issue](https://github.com/hashicorp/terraform-provider-kubernetes-alpha/issues/81) open for it.